src/software/backend/wifi_backend.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "shared/proto/tbots_robot_msg.pb.h"
+#include "shared/proto/tbots_software_msgs.pb.h"
+#include "software/backend/backend.h"
+#include "software/backend/input/network/networking/network_client.h"
+#include "software/networking/proto_multicast_listener.h"
+#include "software/networking/proto_multicast_sender.h"
+
+class WifiBackend : public Backend
+{
+   public:
+    WifiBackend();
```

Comment:
Should the `NetworkConfig` dynamic parameter be passed in here?

Reply:
im holding off on dependency injection until we have the new system if possible

Reply:
Why? The interface is effectively the same no? Might as well do it the "more correct" way with the current system

Reply:
how do we want to go about this? its not super straightforward because of the generic factory. Can we dependency inject and keep the generic factory still generic?

alternatively can have a function to provide the parameters after the backend has been constructed with a `wifi_backend.storeNetworkConfig(config_here)` in `full_system` and inject it like that. 

But that adds custom checks for the backend which kind of sucks and we should avoid if possible.

Thoughts on the best way to do this?

Reply:
Ah this is the same problem as Plays (and the factory). IIRC we considered
* static class functions that return the correct subconfig from some top-level config
* Manually registering things in factories rather than using the generic one

Haven't put a ton of thought into this yet.